pkg/rancher-desktop/integrations/windowsIntegrationManager.ts

```diff
@@ -403,6 +404,33 @@ export default class WindowsIntegrationManager implements IntegrationManager {
     );
   }
 
+  protected async syncHostFile() {
+    await Promise.all(
+      (await this.supportedDistros).map((distro) => {
+        return this.updateHostFile(distro.name);
+      }),
+    );
+  }
+
+  protected async updateHostFile(distro: string) {
+    if (this.settings.experimental?.virtualMachine?.networkingTunnel) {
+      const vethIPAddr = '192.168.1.2';
+      const contents = await fs.promises.readFile(`\\\\wsl$\\${ distro }\\etc\\hosts`, 'utf-8');
+      const lines = contents.split(/\r?\n/g)
+        .filter(line => !line.includes('host.docker.internal'));
+      const hosts = ['host.rancher-desktop.internal', 'host.docker.internal'];
+      const extra = [
+        '# BEGIN Rancher Desktop configuration.',
+        `${ vethIPAddr } ${ hosts.join(' ') }`,
+        '# END Rancher Desktop configuration.',
+      ].map(l => `${ l }\n`).join('');
+
+      await fs.promises.writeFile(
+        `\\\\wsl$\\${ distro }\\etc\\hosts`,
```

Comment:
Hmm, if the default user in a distro is _not_ root, then you won't be able to write to that file over the UNC share. You may need `wsl --distribution … --user 0 --exec` or something along those lines… But then you can't guarantee any particular utility to exist in a distribution either (it could be like a docker container, with just a single executable), so you'd need to run something via `wsl-helper` (via `/mnt/c/…`)